server/mux.go

```diff
@@ -42,13 +43,31 @@ func NewMux(opts MuxOpts, service Service) http.Handler {
 	// Prefix any URLs found in the React assets with any configured basepath
 	prefixedAssets := NewDefaultURLPrefixer(basepath, assets, opts.Logger)
 
+	// Set route prefix for all routes if basepath is present
+	var router chronograf.Router
+	if opts.Basepath != "" && opts.PrefixRoutes {
+		router = &MountableRouter{
+			Prefix:   opts.Basepath,
+			Delegate: hr,
+		}
+	} else {
+		router = hr
+	}
+
 	// Compress the assets with gzip if an accepted encoding
 	compressed := gziphandler.GzipHandler(prefixedAssets)
 
 	// The react application handles all the routing if the server does not
 	// know about the route.  This means that we never have unknown
 	// routes on the server.
-	router.NotFound = compressed
+	hr.NotFound = http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
```

Comment:
Could this logic be combined with `if opts.Basepath != "" && opts.PrefixRoutes {` from line 48 somehow?  Also, let's use http.StripPrefix.